src/monster.cpp

```diff
@@ -138,6 +138,13 @@ monster::monster( const mtype_id &id ) : monster()
         mech_bat_item.ammo_consume( rng( 0, max_charge ), tripoint_zero );
         battery_item = cata::make_value<item>( mech_bat_item );
     }
+
+    loots = type->loots;
+    lootables_requires_all = type->lootables_requires_all;
+    lootable_categories = type->lootable_categories;
+    lootable_materials = type->lootable_materials;
+    lootable_comestibles = type->lootable_comestibles;
+    lootable_itemgroups = type->lootable_itemgroups;
```

Comment:
Why is this data copied into the monster instance? Is it modified there?

Btw. you should move all your looting related data into a separate class, and only store an instance of that class in `mtype`. E.g.
```C++
class lootable {
 std::vector<std::string> categories;
 std::vector<material_id> materials;
 std::vector<std::string> comestibles;
 std::vector<std::string> itemsgroups;
 ...
};
```
Than you can copy them with on statement:
```C++
lootable = type->lootable;
```
And you can extend it easily without changing this copy instruction.

You could also store this as `std::optional<lootable>` in `mtype`, and get rid of the `loots` variable. Instead you check the presence of the `lootable` object in the `std::optional`. If it is there, the monster loots, otherwise it isn't.